### PR TITLE
attributegroup reference skeleton

### PR DESCRIPTION
**Pull request by RFSH**
Related Issue: #482

### CODE REVIEW

File: js/ag_reference.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    
+    constructor: AttributeGroupReference,
+    
+    get location () {
+        return this._location;
+    },
+    
+    /**
+     * the displayname of the reference
+     * TODO not sure if this sis needed
```
Comment (jrchudy) [inline js/ag_reference.js]: We will need it for when there is a picker popup in the facet.

Maybe use the `table.name` and if that's not defined (chances are it won't), we should have fallback heuristics. Maybe displayname should be related to what aggregate was called?

File: js/ag_reference.js [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        if (this._isHTML === undefined) {
+            var value = this.values;
+        }
+        return this._isHTML;
+    },
+
+    get values() {
+        //TODO 
+        // returned value should be the same as current tuple.values 
+        // or we should make this (https://github.com/informatics-isi-edu/ermrestjs/issues/189) change now?
```
Comment (jrchudy) [inline js/ag_reference.js]: I think we do want to move toward this approach with `tuple.values` being an array of objects like:
```
{
  value: <value>,
  isHtml: boolean,
  isMarkdown: boolean
}
```

Karl commented that we should add `isMarkdown`. I'm not sure whether or not he meant to replace `isHtml` with `isMarkdown` or if they should both be defined. I'm trying to think of a case where we define a value as HTML that is not also Markdown.

File: js/ag_reference.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+function AttributeGroupColumn(alias, name, displayname, type, comment, sortable, visible) {
+    /**
+     * @private
+     * @type {string}
+     */
+    this._alias = alias;
+    
+    /**
+     * This might include the aggregate functions,
+     * should we seperate those?
```
Comment (jrchudy) [inline js/ag_reference.js]: Not sure about this.

File: js/ag_reference.js [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        return this._name;
+    },
+    
+    formatvalue: function (data, options) {
+        //TODO should be the same as Column.formatvalue
+        notimplemented();
+    },
+    
+    formatPresentation: function (data, options) {
+        //TODO take cares of markdown_pattern and all, is it needed?
```
Comment (jrchudy) [inline js/ag_reference.js]: I don't know how the user would define a markdown pattern for one of these columns. We would have to design and annotation for it as well.

File: js/ag_reference.js [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    /**
+     * Represents the paging. It will be in the following format:
+     * {"before":boolean, "row":[v1, v2, v3...]}
+     * v1, v2, v3.. are in the same order of columns in the sortObject
+     * 
+     * @type {?Object}
+     */
+    this.pagingObject = paging;
+}
+AttributeGroupLocation.prototype = {
```
Comment (jrchudy) [inline js/ag_reference.js]: Maybe add `get url()`?

File: js/ag_reference.js (hunk 3 above)
Comment (RFSH) [inline js/ag_reference.js]: Use term instead of name.

**Pull request merged**